doc/design.rst

```diff
@@ -209,3 +209,237 @@ _________________
+
+**Rationale/History:**
+
+- Initial decisions aimed to simplify individual adjustments in the projects until we had a better understanding of what needed to be configured.
+- Scattering configuration across various files and tools was a hasty decision to expedite development and accommodate various tools.
+
+**Ideas/Solutions:**
+
+Currently used methods to configure toolbox-based projects:
+
+#. Project configuration: ``noxconfig.py``
+#. Tool-specific configuration files or sections in ``pyproject.toml``
+#. Implementing plugin extension points
+#. Overwriting nox tasks with custom implementations
+#. Replacing with customized workflows of the same name (only applicable for action/workflows)
+
+Refinement:
+
+- Centralize all toolbox based configurations in a toolbox config file (``noxconfig.py``).
+- Rename the toolbox config file from ``noxconfig.py`` to a more appropriate name that reflects its purpose.
+- Document configuration hierarchy and usage.
+
+
+Nox Task Runner
++++++++++++++++
+
+**Description:**
+While Nox isn't a perfect fit, it still meets most of our requirements for a task runner.
+
+**Downsides:**
+
+- Imports over top-level modules are problematic as all contained tasks are imported.
+- Passing and receiving additional arguments to a task is clunky.
+- The default behavior of creating a venv for tasks is undesirable.
+- Nox does not support grouping.
+
+**Rationale/History:**
+
+Why Nox was choosen:
+
+- No Additional Language(s) Required: There was no need to introduce extra programming languages or binaries, simplifying the development process.
+- Python-based: Being Python-based, Nox can be extended and understood by Python developers.
+- Python code: As Nox tasks are defined via Python code, existing scripts can be reused and code can be shared easily.
+- Simplicity: Nox is relatively "small" in functionality, making it somewhat simple to use and understand.
+
+**Ideas/Solutions:**
+
+Grouping:
+
+Since Nox doesn't natively support task grouping, we need a strategy to group commands.
+Therefore, a naming convention to indicate grouping should be adopted.
+
+    Suggestion: Groups will be separated using a :code:`:` (colon) because :code:`-` (dash) might already be used within task names.
+
+Imports:
+
+Consider modularizing tasks to handle top-level imports better.
+
+Others Issues:
+
+Generally, one may consider addressing the other issues by choosing another task runner or creating a small set of CLI tools and extension points manually provided by the toolbox.
+
+
+Poetry for Project Management
++++++++++++++++++++++++++++++
+
+While poetry was and is a good choice for exasol project, dependency, build tool etc. "most recently"
+`uv <https://docs.astral.sh/uv/>`_ has surffeced and made big advanced. Looking at uv it addresses additional itches with
+our projects and therefore in the long run it may be a good idea to migrate our project setups to it.
+Use poetry for project, build and depenency management.
```

Comment:
```suggestion
While poetry was and is a good choice for Exasol project, dependency, build tool etc. "most recently"
`uv <https://docs.astral.sh/uv/>`_ has surfaced and made big advanced. Looking at uv it addresses additional itches with
our projects and therefore in the long run it may be a good idea to migrate our project setups to it.
Use poetry for project, build and dependency management.
```